frame/nomination-pools/src/lib.rs

```diff
@@ -0,0 +1,1967 @@
+		#[pallet::weight(T::WeightInfo::unbond_other())]
+		#[frame_support::transactional]
+		pub fn unbond_other(
+			origin: OriginFor<T>,
+			delegator_account: T::AccountId,
+		) -> DispatchResult {
+			let caller = ensure_signed(origin)?;
+			let (mut delegator, mut bonded_pool, mut reward_pool) =
+				Self::get_delegator_with_pools(&delegator_account)?;
+
+			bonded_pool.ok_to_unbond_other_with(&caller, &delegator_account, &delegator)?;
+
+			// Claim the the payout prior to unbonding. Once the user is unbonding their points
+			// no longer exist in the bonded pool and thus they can no longer claim their payouts.
+			// It is not strictly necessary to claim the rewards, but we do it here for UX.
+			Self::do_reward_payout(
+				&delegator_account,
+				&mut delegator,
+				&mut bonded_pool,
+				&mut reward_pool,
+			)?;
+
+			let balance_to_unbond = bonded_pool.balance_to_unbond(delegator.points);
+
+			// Update the bonded pool. Note that we must do this *after* calculating the balance
+			// to unbond so we have the correct points for the balance:share ratio.
+			bonded_pool.points = bonded_pool.points.saturating_sub(delegator.points);
+
+			// Unbond in the actual underlying pool
+			T::StakingInterface::unbond(bonded_pool.bonded_account(), balance_to_unbond)?;
+
+			let current_era = T::StakingInterface::current_era();
+			let unbond_era = T::StakingInterface::bonding_duration().saturating_add(current_era);
+			// Note that we lazily create the unbonding pools here if they don't already exist
+			let mut sub_pools = SubPoolsStorage::<T>::get(delegator.pool_id)
+				.unwrap_or_default()
+				.maybe_merge_pools(unbond_era);
+
+			// Update the unbond pool associated with the current era with the unbonded funds. Note
+			// that we lazily create the unbond pool if it does not yet exist.
+			if !sub_pools.with_era.contains_key(&unbond_era) {
+				sub_pools
+					.with_era
+					.try_insert(unbond_era, UnbondPool::default())
+					// The above call to `maybe_merge_pools` should ensure there is
+					// always enough space to insert.
+					.defensive_map_err(|_| Error::<T>::DefensiveError)?;
+			}
+			sub_pools
+				.with_era
+				.get_mut(&unbond_era)
+				// The above check ensures the pool exists.
+				.defensive_ok_or_else(|| Error::<T>::DefensiveError)?
+				.issue(balance_to_unbond);
+
+			delegator.unbonding_era = Some(unbond_era);
+
+			Self::deposit_event(Event::<T>::Unbonded {
+				delegator: delegator_account.clone(),
+				pool_id: delegator.pool_id,
+				amount: balance_to_unbond,
+			});
+
+			// Now that we know everything has worked write the items to storage.
+			SubPoolsStorage::insert(&delegator.pool_id, sub_pools);
+			Self::put_delegator_with_pools(&delegator_account, delegator, bonded_pool, reward_pool);
+
+			Ok(())
+		}
+
+		/// Call `withdraw_unbonded` for the pools account. This call can be made by any account.
+		///
+		/// This is useful if their are too many unlocking chunks to call `unbond_other`, and some
```

Comment:
You should say that if error `X` happens in call `unbond_other` then call this function to fix it, and also mention this in the docs of call `unbond_other`.

Reply:
My only hesistation is that the error will come from the staking pallet and it seems out of place to be specifying a staking error in the docs here

Reply:
Well on second thought, we are already mentioning unlocking chunks here, which is already a staking cocept

Reply:
2855e73d244731b86fadf7c62aaabc88f36c8a4a & cd03699793bca9b7f63ef8b8af11dee0956b034a